scripts/generate.mjs

```diff
@@ -0,0 +1,78 @@
+import { Octokit } from 'octokit';
+import * as fs from 'node:fs/promises';
+
+const gh = new Octokit({
+    auth: process.env.GITHUB_TOKEN || process.env.GH_TOKEN
+});
+
+const searchResults = gh.paginate.iterator(gh.rest.search.repos, {
+    q: 'titanium in:topics language:objc language:swift language:java language:kotlin language:javascript',
+    per_page: 100
+});
+
+const repositories = [];
+for await (const {data} of searchResults) {
+    const mappedData = await Promise.all(data.map(mapData));
+    repositories.push(...mappedData.filter(repo => repo));
+}
+
+const out = new URL('../data.json', import.meta.url);
+await fs.writeFile(out, JSON.stringify(repositories, undefined, '\t'));
+
+async function mapData(data) {
+    console.log(`mapping ${data.owner.login}/${data.name}`);
+    const platforms = await getPlatforms(data);
+
+    if (!platforms?.length) {
+        return;
+    }
+
+    return {
+        name: data.name,
+        owner: {
+            login: data.owner.login,
+            html_url: data.owner.html_url,
+        },
+        html_url: data.html_url,
+        description: data.description,
+        updated_at: data.updated_at,
+        platforms
+    }
```

Comment:
Semi-colon, where art thou?

Reply:
I'll get the linting setup to lint this file

Reply:
Added it, had to rejig some stuff but it works (f804a1294ebe4081ef1d99fb5b8a4d3d70b43c44)